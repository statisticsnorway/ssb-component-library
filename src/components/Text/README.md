# Text

> A component for showing Text

### Usage

#### HTML

```html
<span class="ssb-text-wrapper">Default text here</span>

<span class="ssb-text-wrapper small-text">Small text here</span>

<span class="ssb-text-wrapper negative">Negative text here</span>
```

#### React

```jsx harmony
<Text>Default text here</Text>

<Text small>Small text here</Text>

<Text negative>Negative text here</Text>


```

Available props:

| Name      | Type             | Description                |
| --------- | ---------------- | -------------------------- |
| children  | Required node    | Content within the wrapper |
| className | string           | Optional container class   |
| negative  | Optional boolean | Styling for dark theme     |
| small     | Optional boolean | Style textsize             |
