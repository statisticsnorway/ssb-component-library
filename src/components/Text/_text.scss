.ssb-text-wrapper {
  @include open-sans;
  color: $ssb-dark-6;
  font-size: 16px;
  font-style: normal;
  letter-spacing: normal;
  line-height: 1.7;
  max-width: $max-readable-width;

  &.small-text {
    font-size: 14px;
    line-height: 24px;
  }

  &.negative {
    color: $ssb-white;
  }
}
