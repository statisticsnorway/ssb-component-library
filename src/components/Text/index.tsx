import React, { type ReactNode } from 'react'

interface TextProps {
  children: ReactNode
  className?: string
  negative?: boolean
  small?: boolean
}

const Text: React.FC<TextProps> = ({ children, className, negative = false, small = false }) => (
  <span
    className={`ssb-text-wrapper${small ? ' small-text' : ''}${negative ? ' negative' : ''}${className ? ` ${className}` : ''}`}
  >
    {children}
  </span>
)

export default Text
