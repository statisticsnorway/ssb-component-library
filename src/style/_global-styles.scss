body {
  @include open-sans;
  color: $ssb-dark-6;
}

.flex-row {
  display: flex;
  flex-direction: row;
}

.flex-column {
  display: flex;
  flex-direction: column;
}
