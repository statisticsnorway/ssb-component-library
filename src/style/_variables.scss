$ssb-green-1: #e5fee6;
$ssb-green-2: #b6e8b8;
$ssb-green-3: #1a9d49;
$ssb-green-4: #00824d;
$ssb-green-5: #075745;
$ssb-blue-3: #3396d2;
$ssb-red-4: #c4351c;
$negative-red: #f8a67d;
$ssb-dark-2: #a2baba;
$ssb-dark-3: #6f9090;
$ssb-dark-4: #4b7272;
$ssb-dark-5: #274247;
$ssb-dark-6: #162327;
$ssb-white: #fff;
$ice-cold-grey: #ecf8f9;
$ssb-purple-3: #9575ff;


$mobile: 'screen and (max-width: 768px)';
$tablet: 'screen and (max-width: 992px)';
$desktop: 'screen and (min-width: 992px)';

// Use media queries like this:
//@media #{$mobile} { @content }

$input-field-height: 36px;
$global-padding-size: 10px;
$max-content-size: 1200px;

@mixin roboto {
  color: $ssb-dark-6;
  font-family: 'Roboto', sans-serif;
  font-stretch: normal;
  font-weight: normal;
}

@mixin roboto-condenced {
  color: $ssb-dark-6;
  font-family: 'Roboto Condensed', sans-serif;
  font-stretch: condensed;
  font-weight: bold;
}

@mixin open-sans {
  color: $ssb-dark-6;
  font-family: 'Open Sans', sans-serif;
  font-stretch: normal;
  font-weight: normal;
}
