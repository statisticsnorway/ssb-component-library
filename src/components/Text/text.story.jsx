import React from 'react'
import Text from './index'

export default {
  title: 'Text',
}

export const Default = () => <Text>Default text here</Text>

export const Small = () => <Text small>Small text here</Text>

export const Negative = () => <Text negative>Negative text here</Text>
