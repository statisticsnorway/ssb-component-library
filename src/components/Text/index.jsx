import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ssbDark5, ssbDark6, ssbWhite } from '../../style/colors';
import { openSans } from '../../style/mixins';

const Text = ({ children, className, negative, small }) => (
	<TextWrapper
		className={className}
		negative={negative}
		small={small}
	>{ children }
	</TextWrapper>
);

const TextWrapper = styled.span`
	${openSans};
	color: ${props => (props.negative ? ssbWhite : ssbDark6)};
  font-size: ${props => (props.small ? '14px' : '16px')};
  font-style: normal;
  font-weight: normal;
  letter-spacing: normal;
  line-height: 1.7;
`;

Text.defaultProps = {
	className: '',
	negative: false,
	small: false,
};

Text.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	negative: PropTypes.bool,
	small: PropTypes.bool,
};

export default Text;
