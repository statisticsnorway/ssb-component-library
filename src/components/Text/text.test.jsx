import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Text from './index';
import { ssbWhite } from '../../style/colors';

describe('Text component', () => {
	test('Matches the snapshot', () => {
		const component = renderer.create(<Text>Default text here</Text>).toJSON();
		expect(component).toMatchSnapshot();
	});

	test('Changes color when negative', () => {
		const component = renderer.create(<Text negative>Here is a text</Text>).toJSON();
		expect(component).toHaveStyleRule('color', ssbWhite);
	});

	test('Changes font size when `small` is true', () => {
		const component = renderer.create(<Text small>Small text here</Text>).toJSON();
		expect(component).toHaveStyleRule('font-size', '14px');
	});
});
