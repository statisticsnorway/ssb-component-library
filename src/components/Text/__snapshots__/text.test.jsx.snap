// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Matches the snapshot 1`] = `
.c0 {
  font-family: 'Open Sans',sans-serif;
  font-stretch: normal;
  color: #162327;
  font-size: 16px;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.7;
}

<span
  className="c0"
>
  Default text here
</span>
`;
